### DISCUSSION TITLE
Controling FLIR camera setting directly from Bonsai

### DISCUSSION THREAD

**Discussion by hagikent** [bonsai-rx/bonsai#1590] [Q&A]
Hi, 

Some Bonsai-savvy folks (@bruno-f-cruz ?) might have already implemented a workflow to control FLIR camera setting via Bonsai?
(instead of doing it manually in spinview beforehand and save locally on cameras)

Thanks!
Kenta

**Comment by bruno-f-cruz** [comment]
Hey,

You can inherit from the `SpinnakerCapture` class and override the `Configure` method. There is a great implementation example in the following repository:

https://github.com/SainsburyWellcomeCentre/aeon_acquisition/blob/b1ce7074ce366ce91c98b581f919f1b128df67ca/src/Aeon.Acquisition/SpinnakerCapture.cs#L12

You can then use this as an Extension in your workflow, just make sure you add the following line to your `Extensions.cproj` file:

```XML
  <ItemGroup>
    <PackageReference Include="Bonsai.Core" Version="2.8.1" />
    <PackageReference Include="Bonsai.Spinnaker" Version="0.7.1" />
  </ItemGroup>
```
